src/analysis/processing/qgsalgorithmlayoutatlastoimage.cpp

```diff
@@ -141,10 +141,13 @@ QVariantMap QgsLayoutAtlasToImageAlgorithm::processAlgorithm( const QVariantMap
 
     expression = parameterAsString( parameters, QStringLiteral( "FILTER_EXPRESSION" ), context );
     atlas->setFilterExpression( expression, error );
+    atlas->setFilterFeatures( !expression.isEmpty() && error.isEmpty() );
```

Comment:
Isn't this change alone sufficient to fix the issue?

Reply:
This only fixes the atlas to contain a correct value of the features ids (`QgsLayoutAtlas.mFeatureIds`), when exporting the atlas, the issue https://github.com/qgis/QGIS/issues/44242 still persists.

Reply:
I don't think this filterCoverageLayer() / unFilterCoverageLayer() approach s right here. QgsLayoutAtlas' setFilterExpression should suffice to filter atlas features. That's the whole point of this function to begin with.

If there are other lingering issues with setFilterExpression itself, let's nail those.

also, unFilter* would have to be renamed to something like clearFilter* , but I really don't think we need this here.

Reply:
@nyalldawson upon another check, I have seen that this change fixes the https://github.com/qgis/QGIS/issues/44242 issue.